### PR TITLE
Modernize step 32

### PR DESCRIPTION
**Pull request by drwells**
The primary change in this PR is that we can now delete the custom L2 projection code, as `VectorTools::interpolate` has since learned how to do this.

### CODE REVIEW

**Comment by drwells** [comment]
Part of #6774.

File: examples/step-32/doc/intro.dox [hunk 1]
```diff
@@ -1307,8 +1305,8 @@ That said, both step-31 and the current step-32 have not come about by chance
 but are certainly meant as wayposts along the path to a more comprehensive
 program that will simulate convection in the earth mantle. We call this code
 <i>Aspect</i> (short for <i>Advanced %Solver for Problems in Earth's
-ConvecTion</i>); its development is funded by the <a
-href="http://www.geodynamics.org">Computational Infrastructure in
+ConvecTion</i>); its development is funded by
+the <a href="http://www.geodynamics.org">Computational Infrastructure in
 Geodynamics</a> initiative with support from the National Science
-Foundation. We hope to release this code not long after this tutorial program
-will officially be released as part of deal.II 7.1.
```
Comment (kronbichler) [inline examples/step-32/doc/intro.dox]: Ah, what funny how long that text persisted. It's great to do this cleanups from time to time.

**Comment by kronbichler** [comment]
The MPI build including examples succeeded, merging.

**Pull request merged**